framework/src/media/media_player.c

```diff
@@ -74,7 +74,7 @@ play_result_t media_play(int fd, media_format_t format)
 		return PLAY_ERROR_DRIVER;
 	}
 
-	g_pc.buffer_size = pcm_frames_to_bytes(g_pc.pcmout, pcm_get_buffer_size(g_pc.pcmout));
+	g_pc.buffer_size = pcm_get_buffer_size(g_pc.pcmout);
```

Comment:
pcm_get_buffer_size returns # of frames.
Currently default setting is 16bit & stereo.
(Actually what you added in above default config value is set by tinyalsa, not neccessary here.. of course tinyalsa using 1024 frames as a default period size) 
Hence correct buffer size is that 2048 frames * 2 * 2 = 8192 bytes.
Now Audio subsystem using 8192 bytes as a 1 buffer, and there are 8buffers.
There is while loop in read_data_pcm_write which checking readed value, I guess delay comes due to recv size is too small..